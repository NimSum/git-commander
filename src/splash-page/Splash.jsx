import React, { Component } from 'react'

export class Splash extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: ''
    }
  }

  handleChange = e => {
    this.setState({userName: e.target.value});
  }

  startGame = e => {
    e.preventDefault();
    this.props.startGame(this.state.userName);
  }

  render() {
    return (
      <section className="splash-page">
        <h2>GIT COMMANDER</h2>
        <form onSubmit={ this.startGame }>
          <div className="form-item-container">
            <label htmlFor="userName">Your Commander Name:</label>
            <input id="userName" type="text" onChange={this.handleChange} />
            <input className="start-game-btn" type="button" onClick={ this.startGame } value="Let's git started!"/>
          </div>
        </form>
        <img className="octocat-ship" src={require("../images/code-octocat.png")} alt="octocat"/>
      </section>
    )
  }
}

export default Splash
