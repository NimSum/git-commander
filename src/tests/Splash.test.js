import React from 'react';
import Splash from '../splash-page/Splash';
import Adapter from 'enzyme-adapter-react-16';
import { shallow , configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Progress Indicator' , () => {

  let wrapper;
  let mockStartGame = jest.fn();
  let mockEvent = {
    target: { value: 'NIMSUM' },
    preventDefault: () => {}
  }


  beforeEach(() => {
    wrapper = shallow(
      < Splash 
        startGame={mockStartGame}
      />
    );
  });

  it('Should have default states', () => {
    expect(wrapper.state()).toEqual( {
      userName: '',
      octoForward: false
    } )
  })

  it('Should match rendered snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Should handle username change', () => {
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('userName')).toEqual('NIMSUM');
  })

  it('Should start game', () => {
    wrapper.instance().startGame(mockEvent);
    expect(wrapper.state('octoForward')).toEqual(true);
    setTimeout(() => expect(mockStartGame).toHaveBeenCalled(), 2000);
  })

  it('Should invoke startGame when Start Game button is clicked', () => {
    wrapper.find('.start-game-btn').simulate(
      'click', {
        preventDefault: () => {}
      }
    )
  })
  
})