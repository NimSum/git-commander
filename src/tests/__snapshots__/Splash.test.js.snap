// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Indicator Should match rendered snapshot 1`] = `
<section
  className="splash-page"
>
  <h2>
    GIT COMMANDER
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-item-container"
    >
      <label
        htmlFor="userName"
      >
        Your Commander Name:
      </label>
      <input
        autoFocus={true}
        id="userName"
        onChange={[Function]}
        placeholder="Name Required"
        type="text"
      />
      <input
        className="start-game-btn"
        onClick={[Function]}
        type="button"
        value="Start Game"
      />
    </div>
  </form>
  <img
    alt="octocat"
    className="octocat-ship "
    src="jetpacktocat.png"
  />
  <div
    className="instructions-cont"
  >
    <h3>
      Instructions:
    </h3>
    <h4>
      Help octocat make it to remote repo!
    </h4>
    <ol>
      <li>
        Read the prompts for Instructions
      </li>
      <li>
        Type in your command and press enter when ready!
        <p>
          <b>
            Be Careful! You only have one chance per challenge!
          </b>
        </p>
      </li>
      <li>
        Once you reach the 'Remote' Repo, you win! You also have the option to keep going solve harder commands!
      </li>
    </ol>
  </div>
</section>
`;
